tests/unit/fem/test_white_noise.cpp

```diff
@@ -0,0 +1,226 @@
+
+      // Compute mass matrix
+      BilinearForm a(&fespace);
+      a.AddDomainIntegrator(new MassIntegrator());
+      a.Assemble();
+
+      SparseMatrix M;
+      Array<int> empty;
+      a.FormSystemMatrix(empty,M);
+      DenseMatrix Mdense;
+      M.ToDenseMatrix(Mdense);
+
+      // Compare population covariance to mass matrix
+      Mdense -= C;
+      REQUIRE(Mdense.MaxMaxNorm() < 2.0e-3);
+   }
+}
+
+
+#ifdef MFEM_USE_MPI
+
+TEST_CASE("Parallel WhiteGaussianNoiseDomainLFIntegrator on 2D NCMesh",
+          "[Parallel]")
+{
+
+   // Setup
+   const auto order = GENERATE(1, 2, 3);
+   Mesh mesh = Mesh::MakeCartesian2D(3, 3, Element::QUADRILATERAL);
+   // Make the mesh NC
+   mesh.EnsureNCMesh();
+   {
+      Array<int> elements_to_refine(1);
+      elements_to_refine[0] = 1;
+      mesh.GeneralRefinement(elements_to_refine, 1, 0);
+   }
+   ParMesh pmesh(MPI_COMM_WORLD,mesh);
+   mesh.Clear();
+
+   H1_FECollection fec(order, pmesh.Dimension());
+   ParFiniteElementSpace fespace(&pmesh, &fec);
+   HypreParMatrix * P = fespace.Dof_TrueDof_Matrix();
+
+   int ntdofs = fespace.GetTrueVSize();
+   int nvdofs = fespace.GetVSize();
+   ParLinearForm b(&fespace);
+   int seed = 4000;
+   WhiteGaussianNoiseDomainLFIntegrator *WhiteNoise = new
+   WhiteGaussianNoiseDomainLFIntegrator(MPI_COMM_WORLD,seed);
+   WhiteNoise->SaveFactors(pmesh.GetNE());
+   b.AddDomainIntegrator(WhiteNoise);
+   Vector B;
+
+   SECTION("Mean")
+   {
+      // Compute population mean
+      Vector bmean(nvdofs);
+      bmean = 0.0;
+      for (int i = 0; i < N; i++)
+      {
+         b.Assemble();
+         bmean += b;
+      }
+      bmean *= 1.0/(double)N;
+
+      Vector Bmean(ntdofs);
+      P->MultTranspose(bmean,Bmean);
+
+      // Compare population mean to the zero vector
+      REQUIRE(Bmean.Normlinf() < 5.0e-3);
+   }
+
+   SECTION("Covariance")
+   {
+      // Compute population covariance
+      DenseMatrix C(nvdofs);
+      C = 0.;
+      for (int i = 0; i < N; i++)
+      {
+         b.Assemble();
+         AddMultVVt(b, C);
+      }
+      C *= 1.0/(double)N;
+
+      // Create a "sparse" matrix from C;
+      SparseMatrix S(nvdofs);
+      for (int i = 0; i<nvdofs; i++)
+      {
+         for (int j = 0; j<nvdofs; j++)
+         {
+            S.Set(i,j,C(i,j));
+         }
+      }
+      S.Finalize();
+
+      HypreParMatrix * BlockC = new HypreParMatrix(MPI_COMM_WORLD,
+                                                   fespace.GlobalVSize(),
+                                                   fespace.GetDofOffsets(), &S);
+
+      HypreParMatrix * ParC = RAP(BlockC, P);
+      delete BlockC;
+
+      // Compute mass matrix
+      ParBilinearForm a(&fespace);
+      a.AddDomainIntegrator(new MassIntegrator());
+      a.Assemble();
+
+      HypreParMatrix M;
+      Array<int> empty;
+      a.FormSystemMatrix(empty,M);
+
+      // Compare population covariance to mass matrix
+      HypreParMatrix * diff = Add(1., *ParC, -1., M);
```

Comment:
This line causes failures in our nightly tests. The issue is that (at least with older hypre versions) `Add` requires that the two `HypreParMatrix`es have the same `col_map_offd` arrays which in this case is not true.

The alternative that can be used here is:
```suggestion
      M *= -1.;                                                                                                                     
      HypreParMatrix * diff = ParAdd(ParC, &M);                                                                                     
```